fedimint-core/src/util/mod.rs

```diff
@@ -159,6 +159,17 @@ impl SafeUrl {
     pub fn join(&self, input: &str) -> Result<Self, ParseError> {
         self.0.join(input).map(SafeUrl)
     }
+
+    // It can be removed to use `is_onion_address()` implementation,
+    // once https://gitlab.torproject.org/tpo/core/arti/-/merge_requests/2214 lands.
+    #[allow(clippy::case_sensitive_file_extension_comparisons)]
+    pub fn is_onion_address(&self) -> bool {
+        let host = self
+            .host_str()
+            .expect("It should've asserted for `host` on construction");
```

Comment:
How sure are you about that? Couldn't a maliciously crafted (i.e. relative URL) invite code trigger this?

Reply:
Oh, got it! You're correct, a manually crafted invite code could purposely trigger this.

I think returning an error and handling it, would be the proper way to go. Any other thoughts?

Reply:
Actually, let's just return false: if there is no host that's broken and it will fail later anyway.

Reply:
Sure, I did fix in 6df8232eb83e1ae338607c2ed627554bc861af7b

Reply:
https://github.com/fedimint/fedimint/commit/6df8232eb83e1ae338607c2ed627554bc861af7b isn't the cleanest solution (you `unwrap_or_default` an `Option<&str>` instead of an `Option<bool>`, so you get an empty string that you then check the suffix of) but good enough imo.

Reply:
Got it, I can probably make it cleaner in a future PR.